### ISSUE TITLE
ReactGA.initialize is not a function

### ISSUE THREAD

**Issue by joncursi** [react-ga/react-ga#165]
It looks like in `2.3.1`, nothing is being exported by default:

```js
import ReactGA from 'react-ga';

console.log(ReactGA); // {}
```

**Comment by tanqhnguyen** [comment]
Same here, while running test, ReactGA is `{}`

```
import ReactGA from 'react-ga';
console.log(ReactGA); // returns {}
```

**Comment by joevo2** [comment]
Having this issue too

**Comment by thebuilder** [comment]
I'm guessing this might be because Webpack isn't outputting a Library:
https://webpack.js.org/configuration/output/#output-library

**Comment by SimeonC** [comment]
Well damn. My first release as maintainer really was a bad run.
For now please continue using 2.2.0 and I'll get this fixed up ASAP. I've run out of extra time for now, it looks like some things have changed since the last time I was publishing private npm packages using webpack.

Sorry for the problems!!

@thebuilder Thanks for the link!!

**Comment by joevo2** [comment]
2.2.0 no typescript support? :(

**Comment by thebuilder** [comment]
It's great to see an update of the build process for this project, but since it is a library i'm not sure Webpack is the best choice for an NPM build.

You can use Babel directly to compile a release for consumption on NPM (With support for ES2015 modules). Have a look on the `package.json` here: https://github.com/thebuilder/react-module-template/blob/master/package.json

The `scripts > build` and`babel` configuration sections are what you want, along with the updated main and module field:
```
{
  "main": "lib/index.js",
  "module": "es/index.js",
  "jsnext:main": "es/index.js"
}
```

That said, it might still make sense to run it through Webpack to compile a version you can include from CDN. I haven't experimented with this process.

**Comment by gurvinder1129** [comment]
@SimeonC 
In 2.2.0 also, now I am getting the same error.

**Comment by epfisztner** [comment]
@gurvinder1129  v2.2.0 works well. Have you reinstalled the project by package.json?

**Comment by RIP21** [comment]
I'm having this issue even with 2.1.2.
Looks more like sort of an issue of webpack, and bundling of the lib code itself.
Maybe it's better to use just transpiled raw code as an output of the library and not webpack?
Our bundlers are smart enough to figure all imports etc. While maybe for some other like UMD etc. builds you could use webpack etc.

**Comment by drewbietron** [comment]
I was able to get v2.2.0 to work by using the following import.  Will switch back to a proper default import once its updated.

 `import * as ReactGA from 'react-ga'`

**Comment by Nopzen** [comment]
Got the same error as OP when using latest ReactGA.

Followed instructions go to 2.2.0, when using Im getting this error:
<img width="1792" alt="screen shot 2017-09-25 at 18 05 35" src="https://user-images.githubusercontent.com/2326591/30818581-39186e84-a21c-11e7-90d0-100db0c047dc.png">

I've tryed

```
import ReactGA from 'react-ga'
import * as ReactG from 'react-ga
```

Any suggestions?

**Comment by drewbietron** [comment]
@Nopzen - I also have the `"allowSyntheticDefaultImports": true,` option in my tsconfig.  I'd try that with `import * as ReactGA from 'react-ga'`

**Comment by Nopzen** [comment]
@drewbietron I dont run a ts setup :) im runnning the create-react-app boiler plate. aith a little flow on top, but i've ignored all node modules

**Comment by livemixlove** [comment]
Reverting to @2.2.0 I also get `Cannot read property 'Component' of undefined`.  Does something else need to be reverted to an earlier version as well?

**Comment by livemixlove** [comment]
Nevermind.  My mistake.  I hadn't restarted the webpack-dev-server.   Reverting to 2.2.0 does work for me.  @Nopzen  are you using the webpack-dev-server? and if so did you restart it after reverting?

**Comment by Nopzen** [comment]
@livemixlove - Im trying it now!

**Comment by Nopzen** [comment]
@livemixlove Seems to have fixed it! :+1:

**Comment by gkkirsch** [comment]
@SimeonC Hey, we are still getting the `.initialize is not a function` on v2.2.0. We have 2 apps, one using webpack the other using brunch. Any thoughts?

**Comment by siarheipashkevich** [comment]
Same problem here.

**Comment by SimeonC** [comment]
OK, sorry for the problems. Turns out you have to enable `libraryTarget: 'commonjs2'` in webpack or it doesn't do the default export.

On v2.3.2 running ```js
import ReactGA from 'react-ga';
console.log(ReactGA);``` outputs:
```js
{ initialize: [Function: initialize],
  ga: [Function: ga],
  set: [Function: set],
  send: [Function: send],
  pageview: [Function: pageview],
  modalview: [Function: modalview],
  timing: [Function: timing],
  event: [Function: event],
  exception: [Function: exception],
  plugin: { require: [Function: require], execute: [Function: execute] },
  outboundLink: [Function: outboundLink],
  OutboundLink: 
   { [Function: OutboundLink]
     propTypes: 
      { eventLabel: [Function: bound checkType],
        target: [Object],
        to: [Object],
        onClick: [Object] },
     defaultProps: { target: null, to: null, onClick: null },
     trackLink: [Function: bound outboundLink],
     origTrackLink: [Function] } }
```

**Comment by danielg2002** [comment]
@SimeonC May be you should check if typescript definitions file is in the deploy package, also there is a .idea folder (webstorm).

Thanks,
Daniel.

**Comment by SimeonC** [comment]
@danielg2002 Thanks for pointing that out, 2.3.3 has the types now - I accidentally excluded them via the .npmignore file. And I removed the `.idea` folder from the .npmignore file it's in my global .gitignore so I forgot about it when copying over from the .gitignore file. Unfortunately I can't have a global .npmignore file so that will just have to be a bit of extra overhead...

**Comment by gilles-yvetot** [comment]
It is working for me now. Thanks for your React-ivity

**Comment by joevo2** [comment]
So 2.3.3 is good now?

**Comment by gilles-yvetot** [comment]
For me yes.

**Comment by joevo2** [comment]
I'm still having some problem. I'm using typescript but the usual way of doing 
`import * as ReactGA from 'react-ga'` doesn't seems to work and throw this error
```
TypeError: __WEBPACK_IMPORTED_MODULE_17_react_ga__["initialize"] is not a function. (In '__WEBPACK_IMPORTED_MODULE_17_react_ga__["initialize"]('UA-65281289-7')', '__WEBPACK_IMPORTED_MODULE_17_react_ga__["initialize"]' is undefined)
```
instead I did this
`const ReactGA = require('react-ga').default` it works but seems to be weird and dangerous-ish?

**Comment by cheton** [comment]
The output bundle (`dist/react-ga.js`) contains <b>react</b>, <b>prop-types</b>, and <b>object-assign</b> in the same file, and it seems to be unnecessary when setting `libraryTarget: 'commonjs2'` in webpack.

The PR https://github.com/react-ga/react-ga/pull/167 provides a way to exclude dependencies and peerDependencies from the output bundle. It can significantly reduce the footprint size from 185919 to 30785 bytes.

**Comment by danielg2002** [comment]
@joevo2 same here! :(

**Comment by SimeonC** [comment]
@joevo2 @danielg2002 You should be able to use `import ReactGA from 'react-ga'`.
Which is the same as `const ReactGA = require('react-ga').default` which AFAIK is not dangerous, just an older syntax. Let me know how that goes for you (if it doesn't work I'll reopen the issue).

@cheton Thanks for the PR! I'll review it ASAP, have to get back to my day job for today.

**Comment by SimeonC** [comment]
I've published 2.3.4 which will now support both import usages;

1. `import * as ReactGA from 'react-ga';` and `import { OutboundLink } from 'react-ga';`
2. `import ReactGA from 'react-ga';`

**Comment by sonalioswal** [comment]
I am using ngReact and I am getting ReactGA as undefined.
